services/requester/requester.go

```diff
@@ -43,6 +44,7 @@ var (
 
 const minFlowBalance = 2
 const blockGasLimit = 120_000_000
+const txMaxGasLimit = 50_000_000
```

Comment:
I thought max gas was 30M

Reply:
Ethereum has a max gas limit of 30M, per block. However, that is not the case with Flow EVM, we are not bounded by that. This magic value I found by running EVM transactions against Testnet & Mainnet, with `SetComputeLimit(9999)`.

With >= `51M`, we get the following error:
```cadence
error: [Error Code: 1300] evm runtime error: insufficient computation
   --> 8c5303eaa26202d6.EVM:529:8
```

Reply:
The 30M limit on Flow EVM was to keep some room for the Cadence compute units for the Cadence transaction that wraps the EVM transaction. See this FLIP - https://github.com/onflow/flips/blob/main/governance/20240508-computation-limit-hike.md under "Set Gas to compute conversion ratio ...".

Reply:
The weights were set so that 9999 computation equates to 50M evm gas (+- rounding error).
Like Vishal mentioned this was done because we wanted to fit at least 30M evm gas into one flow transaction, but each flow transaction also has a variable computation cost coming form the cadence code (that wraps the EVM transaction). 

I see nothing wrong with adding this check on the gateway, we just have to remember to change it if we change the fee weights.